ci.sh

```diff
@@ -117,13 +117,20 @@ while [ "$#" -gt 0 ]; do
         --no-stress-test )
             NO_STRESS_TEST="True"
         ;;
-        mbed-crypto | pkcs11 | tpm | trusted-service | cryptoauthlib | all)
+        mbed-crypto | pkcs11 | tpm | trusted-service | cryptoauthlib | all | cargo-check)
             if [ -n "$PROVIDER_NAME" ]; then
                 error_msg "Only one provider name must be given"
             fi
             PROVIDER_NAME=$1
-            cp $(pwd)/e2e_tests/provider_cfg/$1/config.toml $CONFIG_PATH
-            if [ "$PROVIDER_NAME" = "all" ]; then
+
+            # If running cargo-check, use all-providers config
+            if [ "$PROVIDER_NAME" = "cargo-check" ]; then
+                cp $(pwd)/e2e_tests/provider_cfg/all/config.toml $CONFIG_PATH
+            else
+                cp $(pwd)/e2e_tests/provider_cfg/$1/config.toml $CONFIG_PATH
+            fi
```

Comment:
This shouldn't be needed as during the `cargo-check` tests we do not run Parsec but only build it, the config file is only useful at runtime 👌

Reply:
Hmm, true, I suggested this because the `trusted-service-provider` would need setting up for those tests, but I didn't realise it's disabled for now.

Reply:
Best to leave it in or remove it do you reckon?
If it is going to be required in the future it's an insignificant instruction in terms of test duration, so leaving it in wouldn't bring any adverse performance to the tests.

Reply:
I'd say remove it - there's plenty going on in this file, no reason to "maintain" things that aren't even used.